distccd: add page (#5912)
tldr-pages/tldr: pages/common/distccd.md

# distccd

> Server daemon for distcc distributed compiler.

> > More information: <https://distcc.github.io/>.

Reviewer: ```suggestion
> More information: <https://distcc.github.io>.
```

> - Start a daemon with default settings:

Reviewer: Can you run multiple daemons? If not, I think `Start the daemon ...` would be more fitting.

Author: Technically it is possible to start a few daemons, each listening to a different port. No idea why would anyone want that, but I wouldn't imply it is not possible.

> - Start a daemon with default settings:

Reviewer 2: ```suggestion
- Start a daemon with the default settings:
```

`distccd --daemon`

> - Start a daemon accepting connections from private network ipv4 ranges:

Reviewer: ```suggestion
- Start a daemon accepting connections from private network IPv4 ranges:
```

> - Start a daemon, accepting connections from private network IPv4 ranges:

Reviewer 2: ```suggestion
- Start a daemon, accepting connections from private IPv4 ranges:
```

Author: I'm not entirely sure. Here's my [source](https://wiki.archlinux.org/title/Distcc#Volunteers):

> At a minimum, add the --allow-private switch which covers a number of ipv4 private network ranges.

Perhaps that would be the preferred form?

```suggestion
- Start a daemon, accepting connections from IPv4 private network ranges:
```

`distccd --daemon --allow-private`

- Start a daemon, accepting connections from a specific network address or address range:

> `distccd --daemon --allow {{IP[/BITS]}}`

Reviewer 3: We don´t use this notation `IP[/BITS]`, we use `lower_snake_case` and ORs `|` for argument options:
```suggestion
`distccd --daemon --allow {{ip_address|netmask_address}}`
```

Author: This bit is taken straight from `distccd --help` output.
`    -a, --allow IP[/BITS]      client address access control`
I thought the original output would be clearer, but if it violates tldr guidelines, I am in no opposition.

Reviewer 3: :thinking: With your example I think I can enter a specific IP like `192.168.0.4` or a mask like `192.168.0.0/24`. Is this correct?

Author: Yes, that's my understanding as well. Personally my knowledge about networking is quite limited, and perhaps that's the reason why `IP[/BITS]` is easier for me to understand.
We may settle this by splitting this example into two. One example for a specific IP, and another for IP with BITS. ;)

Reviewer 4: Sounds good, @tomek-szczesny!

Indeed, a netmask is not what we're after here - an example of a netmask is `255.255.255.0`.

> - Start a daemon with maximum of 4 tasks at any time and lowered priority (niceness = 5):

Reviewer: ```suggestion
- Start a daemon with a maximum of 4 tasks at any time and lowered priority (niceness = 5):
```

> - Start a daemon with a maximum of 4 tasks at any time and lowered priority (niceness = 5):

Reviewer 2: `niceness = 5` seems redundant to me, because of `--nice 5`
```suggestion
- Start a daemon with lowered priority that can run a maximum of 4 tasks at a time:
```

> `distccd --daemon --jobs {{4}} --nice {{5}}`

Reviewer 5: I think we should either use hard-coded values:
```
`distccd --daemon --jobs 4 --nice 5`
```
or variables:
```
`distccd --daemon --jobs {{maximum_job_number}} --nice {{no_idea_how_to_name_it_but_I_am_sure_you_know}}`
```

Author: I experimented with a few formats of this example and found this one to be the cleanest for quick reference.
Also, the [style guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md) says:

> If the example is clearer with an actual value rather than a generic placeholder, use the actual value. For example, use iostat {{2}} rather than iostat {{interval_in_secs}}.

Reviewer: I think we can keep it like that. One thing we could do is change it to maximum priority. Right now, I don't have a feeling for what the niceness really is. If I knew the maximum value, it would be easier to come up with a lower one. (assuming it goes from 0 to max)

Reviewer: I didn't know it was that complicated. In that case I think you got a point with 
> For those who don't know how to handle niceness, there is a ready example how to reduce priority of a daemon below "normal". For those who know what niceness is, that's a clear indication that they can put any valid value they want.

Maybe we should get some other opinions (@bl-ue, @navarroaxel, @nicokosi)

> - Start a daemon with a maximum of 4 tasks at any time and lowered priority (niceness = 5):
> 
> `distccd --daemon --jobs {{4}} --nice {{5}}`

Author: Suggested modification for a point risen by @marchersimon:

```suggestion
- Start a daemon with a maximum of 4 tasks at any time and the lowest priority (niceness = 19):

`distccd --daemon --jobs {{4}} --nice {{19}}`
```

Author: I really don't think a novice user needs to know all that. Any priority lower than normal will be fine for most user cases. How often do you adjust priority of the processes anyway? ;)

Reviewer: Since this change turns out to complicate the example quite a bit, I think we should keep the original.

> - Start a daemon and register via mDNS/DNS-SD (Zeroconf):

Reviewer 2: ```suggestion
- Start a daemon and register it via mDNS/DNS-SD (Zeroconf):
```
